doc/rvv-intrinsic-spec.adoc

```diff
@@ -316,202 +351,362 @@ The following intrinsics have an irregular naming pattern.
+
+[[pseudo-vsetvl]]
+=== `vsetvl`
+
+The `vsetvl` intrinsics return the number of elements processed in a stripmining loop when  provided with the element width and LMUL in the intrinsic suffix.
+
+NOTE: The implementation must respect the ratio between SEW and LMUL given to the intrinsic. On the other hand, as mentioned in <<control-of-vl>>, the `vsetvl` intrinsics do not necessarily map to the emission a `vsetvl` instruction of that exact SEW and LMUL provided. The actual value written to the `vl` control register is an implementation defined behavior and typically not known until runtime.
+
+[[pseudo-vsetvlmax]]
+=== `vsetvlmax`
+
+The `vsetvlmax` intrinsics return `VLMAX` ^5^ when provided with the element width and LMUL in the intrinsic suffix.
 
-[[pseudo-vsetvl-vsetvlmax]]
-=== `vsetvl`/`vsetvlmax`
+NOTE: As mentioned in <<control-of-vl>>, the `vsetvlmax` intrinsics do not necessarily map to the emission a `vsetvl` instruction of that exact SEW and LMUL provided. The actual value written to the `vl` control register is an implementation defined behavior and typically not known until runtime.
 
 [[pseudo-vreinterpret]]
 === `vreinterpret`
 
+The `vreinterpret` intrinsics are provided for users to transition across the strongly-typed scheme. The intrinsic is limited to conversion between types operating upon the same number of registers.
+
+The intrinsics do not alter values held within. Please use `vfcvt/v(f)wcvt/v(f)ncvt` intrinsics if you seek to extend, narrow, or perform real float/interger type conversions for the values.
+
 [[pseudo-vundefined]]
 === `vundefined`
+The `vundefined` intrinsics are placeholders for `vset` and `vcreate` to represent unspecified values.
+
+[[pseudo-vget]]
+=== `vget`
+
+The `vget` intrinsics allow users to obtain small LMUL values from larger LMUL ones. The `vget` intrinsics also allows users to extract non-tuple (`NFIELD=1`) types from tuple (`NFIELD>1`) types after segment load intrinsics.
+
+The intrinsics do not map to any real instruction. Whether if the implementation will generate vector move instructions is an optimization issue.
 
-[[pseudo-vget-vset]]
-=== `vget`/`vset`
+[[pseudo-vset]]
+=== `vset`
 
-[[pseudo-vlmul_trunc-vlmul_ext]]
-=== `vlmul_trunc`/`vlmul_ext`
+The `vset` intrinsics allow users to combine small LMUL values into larger LMUL ones. The `vset` intrinsics also allows users to combine non-tuple (`NFIELD=1`) types to tuple (`NFIELD>1`) types for segment store intrinsics.
+
+The intrinsics do not map to any real instruction. Whether if the implementation will generate vector move instructions is an optimization issue.
+
+[[pseudo-vlmul_trunc]]
+=== `vlmul_trunc`
+
+The `vlmul_trunc` intrinsics are syntactic sugars that essentially have the same semantic as `vget` with `idx=0`.
```

Comment:
This is style probably: the "essentially" may suggest that there is some slight difference (even if small). If the semantics is exactly the same due to the syntactic sugar nature of those intrinsics, I'd drop the word.